src/neo4j_genai/neo4j_queries.py

```diff
@@ -40,6 +40,13 @@
     "YIELD node, score"
 )
 
+MATCH_QUERY = (
```

Comment:
Do we expert users to use more customized queries, especially using Cypher pattern like for instance:
```
...
MATCH (node:Employee)<-[:BELONGS_TO]-( :Department {name: "some_values_from_filter"})
...
```
?

My concern is: should this query be part of the Retriever class instead of being abstracted here?

Reply:
Not sure I understand exactly what you mean, but I this does look a bit strange actually. It's very specific and open-ended at the same time.
All external retrievers would have the same need, but it's just a single pretty simple query so no real need to provide  an abstraction for it.

I'll change it and move the query into the weaviate retriever.